range_sensor_broadcaster/test/test_range_sensor_broadcaster.cpp

```diff
@@ -0,0 +1,253 @@
+{
+  ASSERT_THROW(init_broadcaster(""), std::exception);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Initialize_RangeBroadcaster_Success)
+{
+  ASSERT_EQ(
+    init_broadcaster("test_range_sensor_broadcaster"), controller_interface::return_type::OK);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Configure_RangeBroadcaster_Error_1)
+{
+  // First Test without frame_id ERROR Expected
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  std::vector<rclcpp::Parameter> parameters;
+  // explicitly give an empty sensor name to generate an error
+  parameters.emplace_back(rclcpp::Parameter("sensor_name", ""));
+  ASSERT_EQ(configure_broadcaster(parameters), controller_interface::CallbackReturn::ERROR);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Configure_RangeBroadcaster_Error_2)
+{
+  // Second Test without sensor_name ERROR Expected
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  std::vector<rclcpp::Parameter> parameters;
+  // explicitly give an empty frame_id to generate an error
+  parameters.emplace_back(rclcpp::Parameter("frame_id", ""));
+  ASSERT_EQ(configure_broadcaster(parameters), controller_interface::CallbackReturn::ERROR);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Configure_RangeBroadcaster_Success)
+{
+  // Third Test without sensor_name SUCCESS Expected
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  ASSERT_EQ(
+    range_broadcaster_->on_configure(rclcpp_lifecycle::State()),
+    controller_interface::CallbackReturn::SUCCESS);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Activate_RangeBroadcaster_Success)
+{
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  range_broadcaster_->on_configure(rclcpp_lifecycle::State());
+
+  ASSERT_EQ(
+    range_broadcaster_->on_activate(rclcpp_lifecycle::State()),
+    controller_interface::CallbackReturn::SUCCESS);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Update_RangeBroadcaster_Success)
+{
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  range_broadcaster_->on_configure(rclcpp_lifecycle::State());
+  ASSERT_EQ(
+    range_broadcaster_->on_activate(rclcpp_lifecycle::State()),
+    controller_interface::CallbackReturn::SUCCESS);
+  // range_broadcaster_->get_node()->get_clock()->now();
+  // auto result = range_broadcaster_->update(rclcpp::Time(0.0),
+  // rclcpp::Duration::from_seconds(0.01));
+  auto result = range_broadcaster_->update(
+    range_broadcaster_->get_node()->get_clock()->now(), rclcpp::Duration::from_seconds(0.01));
+
+  ASSERT_EQ(result, controller_interface::return_type::OK);
+}
+
+TEST_F(RangeSensorBroadcasterTest, Publish_RangeBroadcaster_Success)
+{
+  init_broadcaster("test_range_sensor_broadcaster");
+
+  range_broadcaster_->on_configure(rclcpp_lifecycle::State());
+  range_broadcaster_->on_activate(rclcpp_lifecycle::State());
+
+  sensor_msgs::msg::Range range_msg;
+  subscribe_and_get_message(range_msg);
+
+  EXPECT_EQ(range_msg.header.frame_id, frame_id_);
+  EXPECT_PRED_FORMAT2(::testing::FloatLE, range_msg.range, sensor_range_);
```

Comment:
```suggestion
  EXPECT_THAT(range_msg.range, ::testing::Le(sensor_range_));
```
is this accurate to what you meant here?

Reply:
Ok, yes I guess you are right this is the same.
In the mean time I read the [matchers.md](https://github.com/google/googletest/blob/main/docs/reference/matchers.md)

I what I really want to test is the following :

`EXPECT_THAT(range_msg.range, ::testing::FloatEq(sensor_range_));`

So I will change all the occurrences and commit again + rebase

Reply:
Do you agree with the changes?